include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -183,17 +183,20 @@ class argument_parser
                     char const * const * const  argv,
                     bool version_check = true,
                     std::vector<std::string> subcommands = {}) :
-        version_check_dev_decision{version_check}
+        version_check_dev_decision{version_check},
+        subcommands{std::move(subcommands)}
     {
         if (!std::regex_match(app_name, app_name_regex))
-            throw design_error{"The application name must only contain alpha-numeric characters "
-                                               "or '_' and '-' (regex: \"^[a-zA-Z0-9_-]+$\")."};
-        for (auto & sub : subcommands)
+            throw design_error{"The application name must only contain alpha-numeric characters or '_' and '-' "
+                               "(regex: \"^[a-zA-Z0-9_-]+$\")."};
+
+        for (auto & sub : this->subcommands)
             if (!std::regex_match(sub, std::regex{"^[a-zA-Z0-9_]+$"}))
```

Comment:
What is the difference between the app_name_regex and this one? And why does it have different requirements then the app name? For Example no `-` sign?

Reply:
Let me check..
So the subcommand parser name is automatically build with `appname``-``subcommand`, that's why the `-`. I don't remember adding the restriction but it was most probably me. I currently see no reason why this shouldn't be allowed.
I will fix that in a follow up PR though

Reply:
So the design is that the subcommand parser needs to be called `foo-bar` insteaf of `foo bar`?

Reply:
no the app is still called with `./foo bar` on the command line. But the app name internally `subparser.info.app_name` is called `foo-bar`. A name is required and for the subparser I set the name internally, I guess I chose to add the top level parser to it :shrug: could disambiguate stuff in the future or with multiple nested parsers.

Reply:
I added a follow up issue to track this https://github.com/seqan/product_backlog/issues/234